### PR TITLE
581 multiple auth group groups

### LINKED ISSUE #581: Allow users to belong to multiple groups
PLUTO assumes a user only belongs to a single authentication group.  However there are cases where a person may belong to multiple groups and wish to modify all validation rules owned by multiple groups (without being an admin).  Please allow users to pass in a list of groups in the AUTH-GROUP header (perhaps delimited by ";").

### PR DESCRIPTION
**Pull request by bb-acarrera**
This adds support for multiple owner groups sent in the AUTH-GROUP header as a ";" list.  

Resolves #581

### CODE REVIEW

File: src/server/tests/unit/test-rulesets.js [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+QUnit.test( "validateInputOwnerGroup: Admin Auth group with mult group and no input group provided", function(assert){
+  let auth = {
+    admin: true,
+    group: "GroupA;GroupB"
+  }
+  let rv = rulesetRouter.validateInputOwnerGroup(auth, undefined);
+
+  assert.equal(true, rv.valid);
+  assert.equal(undefined, rv.ownergroup);
+});
```
Comment (gfiorav) [inline src/server/tests/unit/test-rulesets.js]: Amazing suite. Kudos!

One caveat: try using `deepEqual` instead of `equal`, the reason being that it `equal` only compares the first level of an object (!!!), given way to false positives. Not relevant in this suite, since you compare literals.

Comment (gfiorav) [reply src/server/tests/unit/test-rulesets.js]: Also, I'm missing a test on expected behaviour if `group` were to be `""` or `admin` be `null` or `undefined`.

Just to have a clear idea of what to expect.

File: src/common/authUser.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+  get groupsAsString() {
+    return this._groups.join(';');
+  }
+
+  get admin() {
+    return this._admin;
+  }
+
+	validateInputOwnerGroupEdit(inputgroup) {
```
Comment (gfiorav) [inline src/common/authUser.js]: This indentation change feels weird!

File: src/common/authUser.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	validateInputOwnerGroup(inputgroup) {
+		// Always allow an admin to overwrite the group
+		if(this.admin) {
+			return {
+				valid: true,
+				ownergroup: inputgroup
+			};
+    }
+
+    if(!inputgroup || inputgroup === "") {
```
Comment (gfiorav) [inline src/common/authUser.js]: Id use parenthesis to enforce precedence... makes it more readable too

To the untrained eye, this could be:
```js
((!inputgroup || inputgroup) === "")
```

Which is not what you want!
